### PR TITLE
Throw an error when oid is missing

### LINKED ISSUE #128: please throw when oid is nullish
Hi

i just had a hard time debugging a stupid mistake:
```
pg.types.setTypeParser(
  pg.types.builtins.TIMESTAMPZ,
  str => myDateParser(str)
);
```

which silently fails to call myDateParser.
Can you spot the mistake ?
It took me a while to get it... so i think it would be nice if `setTypeParser(null, ...)` threw an error.

### PR DESCRIPTION
**Pull request by kapouer**
Fixes #128

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ function setTypeParser (oid, format, parseFn) {
     parseFn = format
     format = 'text'
   }
+  if (oid == null) throw new Error("Missing oid");
```
Comment (sehrope) [inline index.js:36]: How about explicitly checking that oid is an integer?

```js
if (!Number.isInteger(oid)) {
    throw new Error('oid must be an integer: ' + oid)
}
```

Comment (kapouer) [reply index.js]: Sure ! Meanwhile i fixed style and added a basic test.

Comment (sehrope) [reply index.js]: Could use `TypeError` rather than a vanilla Error too.

**Comment by kapouer** [comment]
I'm not sure how to correctly use tape to test that the thrown error is TypeError

**Comment by sehrope** [comment]
You can use a regex in the second arg of `t.throws(...)` to validate the thrown error:

```js
t.test('setTypeParser should throw when oid is not an integer', function (t) {
    t.throws(function () {
      setTypeParser(null, function () {})
    }, /^TypeError: oid must be an integer/)    // <-- Add this
    t.throws(function () {
      setTypeParser('a', function () {})
    }, /^TypeError: oid must be an integer/)    // <-- Add this
    t.end()
  })
```

**Comment by kapouer** [comment]
Ha, ok. Done and force-pushed.

**Comment by bendrucker** [comment]
Perfect, thanks!

**Comment by sehrope** [comment]
LGTM.

@bendrucker Should add a note that the only thing this might break is someone setting a parser using the string representation of an int for the oid value. That would have worked before as the prop lookup of the oid gets stringified but I don't think it's a big deal to break that use case. A note should be fine as it's undocumented behavior and anyone would see it at app startup too.

**Comment by bendrucker** [comment]
Good catch! This is going to go out in a major release that's months overdue here anyway. Will make sure to note it in the main pg changelog.

**Pull request merged**